src/pages/commits/hooks/useCommitsLimit.ts

```diff
@@ -0,0 +1,25 @@
+import { MutableRefObject, useEffect, useRef } from "react";
+import { useDimensions } from "hooks/useDimensions";
+import { useQueryParams } from "hooks/useQueryParam";
+import { MainlineCommitQueryParams } from "types/commits";
+
+export const useCommitsLimit = <T extends HTMLElement>(): [
+  MutableRefObject<T>,
+  number
+] => {
+  const [params, setParams] = useQueryParams();
+  const commitsContainerRef = useRef<T>();
+  const { width } = useDimensions(commitsContainerRef);
+  const nextLimit = Math.max(Math.round(width / COL_WIDTH), MIN_LIMIT);
+
+  useEffect(() => {
+    const nextParams = { ...params };
+    delete nextParams[MainlineCommitQueryParams.SkipOrderNumber];
+    setParams(nextParams);
```

Comment:
I would recommend `useQueryParam` instead of `useQueryParams`  the distinction is that we only need to update the SkipOrderNumber param

const [skipOrderNumber
```suggestion
  const [, setSkipOrderNumber] = useQueryParam(
    MainlineCommitQueryParams.SkipOrderNumber,
    0
  );
  const commitsContainerRef = useRef<T>();
  const { width } = useDimensions(commitsContainerRef);
  const nextLimit = Math.max(Math.round(width / COL_WIDTH), MIN_LIMIT);

  useEffect(() => {
    setSkipOrderNumber(undefined);
  }, [nextLimit]); // eslint-disable-line react-hooks/exhaustive-deps


```